workers/data_refinery_workers/processors/smasher.py

```diff
@@ -502,10 +513,18 @@ def _smash(job_context: Dict, how="inner") -> Dict:
 
         # Once again, `key` is either a species name or an experiment accession
         for key, input_files in job_context['input_files'].items():
+            # Check if we need to copy the quant.sf files
+            if job_context['dataset'].quant_sf_only:
+                outfile_dir = smash_path + key + "/"
+                os.makedirs(outfile_dir, exist_ok=True)
+                sync_quant_files(outfile_dir, input_files, job_context)
+                # we ONLY want to give quant sf files to the user if that's what they requested
+                continue
```

Comment:
Hmm, have you fully considered the implications of terminating this loop here? I can at least tell that `num_samples` will stay at 0, so we'll report to the user in the metadata that we've served 0 samples regardless of what we've actually given them. There's probably others, but even that one alone is enough to make me think the metadata might not look so good.